packages/jest-emotion/src/index.js

```diff
@@ -56,43 +56,76 @@ type Options = {
   DOMElements?: boolean
 }
 
-function filterEmotionProps(props = {}) {
+function filterEmotionProps(props = {}, shouldKnowStyles = true) {
   const {
     css,
     __EMOTION_TYPE_PLEASE_DO_NOT_USE__,
     __EMOTION_LABEL_PLEASE_DO_NOT_USE__,
     ...rest
   } = props
 
-  rest.css = 'unknown styles'
+  if (shouldKnowStyles) {
+    rest.css = 'unknown styles'
+  }
 
   return rest
 }
 
+function isShallowEnzymeElement(element, classNames) {
+  let renderedClassNames = (element.children || [])
+    .map(({ props = {} }) => props.className)
+    .filter(Boolean)
+    .join(' ')
+  return !classNames.some(className =>
+    new RegExp(className).test(renderedClassNames)
+  )
+}
+
 export function createSerializer({
   classNameReplacer,
   DOMElements = true
 }: Options = {}) {
   let cache = new WeakSet()
   function print(val: *, printer: Function) {
-    if (isEmotionCssPropEnzymeElement(val)) {
-      return val.children.map(printer).join('\n')
-    }
+    let elements = getStyleElements()
+    let keys = getKeys(elements)
     if (isEmotionCssPropElementType(val)) {
       return printer({
         ...val,
         props: filterEmotionProps(val.props),
         type: val.props.__EMOTION_TYPE_PLEASE_DO_NOT_USE__
       })
     }
+    if (isEmotionCssPropEnzymeElement(val)) {
+      let expectedClassNames = (val.props.css.name || '')
+        .split(' ')
+        .map(className => keys.map(key => `${key}-${className}`))
+        .reduce((flat, values) => flat.concat(values), [])
```

Comment:
there is already a `flatMap` in `jest-emotion`, could be reused here
https://github.com/emotion-js/emotion/blob/e20cf6ad6133db82d9991ef354e01811e20aedc5/packages/jest-emotion/src/utils.js#L3

Reply:
thanks!